internal/collections/concat.go

```diff
@@ -76,35 +73,44 @@ func (c *ConcatKeyed) Get(key string) []string {
 // FindRegex returns a slice of MatchData for the regex
 func (c *ConcatKeyed) FindRegex(key *regexp.Regexp) []types.MatchData {
 	var res []types.MatchData
-	for _, c := range c.data {
-		res = append(res, c.FindRegex(key)...)
+	for _, d := range c.data {
+		res = append(res, replaceVariable(c.variable, d.FindRegex(key))...)
 	}
 	return res
 }
 
 // FindString returns a slice of MatchData for the string
 func (c *ConcatKeyed) FindString(key string) []types.MatchData {
 	var res []types.MatchData
-	for _, c := range c.data {
-		res = append(res, c.FindString(key)...)
+	for _, d := range c.data {
+		res = append(res, replaceVariable(c.variable, d.FindString(key))...)
 	}
 	return res
 }
 
 // FindAll returns all matches for all collections
 func (c *ConcatKeyed) FindAll() []types.MatchData {
 	var res []types.MatchData
-	for _, c := range c.data {
-		res = append(res, c.FindAll()...)
+	for _, d := range c.data {
+		res = append(res, replaceVariable(c.variable, d.FindAll())...)
 	}
 	return res
 }
 
 // Name returns the name for the current CollectionconcatCollection
 func (c *ConcatKeyed) Name() string {
-	return c.name
+	return c.variable.Name()
 }
 
 // Reset the current ConcatMap
 func (c *ConcatKeyed) Reset() {
 }
+
+// replaceVariable ensures a returned match references the variable of a concatenated variable,
```

Comment:
This was a bug as far as I can tell since the proxy collection days

https://github.com/corazawaf/coraza/pull/606/files#diff-00c784cca0e59aaa33d46c2e6202b3934c38037b006cc25fcd0184e72c60829bL24

Something about the rest of this change caused a test to fail because of it but I couldn't figure out what...

Reply:
BTW the test is this one

https://github.com/corazawaf/coraza/blob/v3/dev/testing/engine/match.go#L62

Without this fix, matched_var_name had `ARGS_GET:id`, not `ARGS:id`

/cc @M4tteoP